### ISSUE TITLE
Add AI.CLASSIFY and constrained decode support

### ISSUE THREAD

**Issue by shreyashankar** [fsdatalab/quail#105]
## Problem

Quail currently supports AI operators that decide between `TRUE` and `FALSE` without generating answer tokens.

Add `AI.CLASSIFY`, which returns one category from a list supplied by the user. A category may contain several tokens, so the execution engine must support decode. Decode means generating output one token at a time after processing the input prompt.

The first version should return exactly one category. Multiple categories, `AI.EXTRACT`, and `AI.MAP` are out of scope.

## SQL interface

A user can project a category into the result:

```sql
SELECT
    t.id,
    AI.CLASSIFY(
        PROMPT('Classify this support request:\n\n{0}', t.body),
        ['billing', 'shipping', 'product', 'other']
    ) AS category
FROM tickets AS t;
```

A user can compare the category in `WHERE`:

```sql
SELECT t.id
FROM tickets AS t
WHERE AI.CLASSIFY(
    PROMPT('Classify this support request:\n\n{0}', t.body),
    ['billing', 'shipping', 'product', 'other']
) = 'billing';
```

A user can use a classification over two relations as a join predicate:

```sql
SELECT r.id, d.id
FROM reports AS r
JOIN diagnoses AS d
  ON AI.CLASSIFY(
       PROMPT(
         'Classify the relationship between report {0} and diagnosis {1}.',
         r.body,
         d.description
       ),
       ['match', 'no_match']
     ) = 'match';
```

The function returns a `STRING` that exactly matches one supplied category. The categories must be known while planning the query. An empty category list, duplicate categories, or a comparison with a value outside the category list should produce a clear compile error.

The interface follows the single category form of [BigQuery `AI.CLASSIFY`](https://docs.cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-ai-classify).

## Query planning

The logical plan should represent `AI.CLASSIFY` as an AI operator that returns a string. It must retain the prompt and the allowed categories.

The planner should inspect which relations the prompt references. A call that references one relation classifies each input row. A call that references two relations classifies candidate document pairs.

A projected classification should run after filters and joins that do not depend on its result. Delaying the classification reduces the number of model evaluations.

A classification used in `WHERE` must run before the comparison can filter rows. The planner should use its estimated cost and selectivity when ordering it with other predicates.

A classification used in `JOIN ... ON` becomes an AI join operation over candidate pairs. The planner should include the category comparison when estimating join selectivity.

If the same `AI.CLASSIFY` call appears in both a condition and a projection, the planner should evaluate it once. It should retain the category for rows that pass the condition and reuse it in the final projection.

## Decode planning

The planner must account for both input processing and generated category tokens.

The batch size must leave enough GPU memory for every input to generate the longest allowed category. The time estimate should use the expected category length. Until category frequencies are available, the planner can use the mean tokenized category length and show that assumption in `query.explain()`.

The cost model must count each generated token and its attention work. It must also account for any KV retained while the engine generates the category.

`query.explain()` should show the allowed categories, their token lengths, the expected output length, the estimated number of evaluations, and the estimated time.

## Execution

The Quail backend should decode only valid categories. It should build a token prefix tree from the allowed categories and restrict each decode step to tokens that can complete one of them.

The decoder must handle categories that share a token prefix. For example, `sport` and `sports news` must remain valid until enough tokens distinguish them.

The runtime should return the original category string, not a decoded string that merely resembles it. The runtime must stop after completing one allowed category.

The result collector should return projected categories as a nonnullable Arrow string column. A condition should use the category for comparison without adding it to the result unless the user also selects it.

## Model support

Every session must continue to name its model and device through `EngineConfig`. Do not add a default model or device.

The first implementation should use a supported Qwen3 model. The model adapter must expose full vocabulary logits during decode instead of the current output head restricted to `TRUE` and `FALSE`.

## Results and metrics

The query report should include generated tokens as well as fresh input tokens. It should also report query time, throughput, and GPU cost per query.

A filter query should report documents per second. A join query should report document pairs per second.

## Acceptance criteria

- `AI.CLASSIFY` returns exactly one supplied category.
- Categories that contain several tokens work.
- Categories with a shared token prefix work.
- Projection returns the category as an Arrow string column.
- Equality comparisons work in `WHERE`.
- Equality comparisons work in `JOIN ... ON`.
- The planner delays a projection only classification until after independent filters and joins.
- The planner evaluates a repeated classification once when a condition and projection use the same call.
- The batch size includes the maximum category length.
- The cost model includes generated tokens and decode attention.
- `query.explain()` shows the category and decode estimates.
- Tests cover parsing, planning, constrained decode, result collection, and invalid category lists.
- One Modal filter run and one Modal join run verify the implementation.
- Each Modal run states its prediction first and reports query time, throughput, and GPU cost per query.
